### PR TITLE
fix typedoc generation

### PR DESCRIPTION
**Pull request by paul-marechal**
`typedoc@0.20.36` seems like the last version supporting `typescript@3`.

Since first introduced, TypeDoc options changed and the repository is
build in a way that doesn't work well for it anymore.

Pin to `typedoc@0.20.36`.
Update `configs/typedoc.json` options.
Re-enable CI step to create the TypeDoc.

#### How to test

Optional: set `NODE_OPTIONS=--max-old-space-size=8192` to avoid Node's OOM error.
`yarn && yarn docs` then open `gh-pages/index.html` in your browser to navigate the generated docs.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -102,11 +102,20 @@ jobs:
       - name: Pre-npm-Publish
         run: |
           yarn --skip-integrity-check --network-timeout 100000
-          yarn compile
```
Comment (marcdumais-work) [inline .github/workflows/build.yml:105]: not needed?

Comment (paul-marechal) [reply .github/workflows/build.yml:105]: Not needed: This script is part of of `prepare` being run after `yarn install` ✔️

**Comment by marcdumais-work** [comment]
It fails for me locally, with a heap out of memory exception: 

```
$ yarn docs
yarn run v1.22.10
$ rimraf gh-pages/docs/next && typedoc --tsconfig tsconfig.json --options configs/typedoc.json .
Info: Loaded plugin /home/lmcmcds/theia/node_modules/typedoc-plugin-external-module-map

<--- Last few GCs --->

[22685:0x4126fd0]    34767 ms: Scavenge 2044.4 (2051.1) -> 2043.8 (2051.6) MB, 5.5 / 0.0 ms  (average mu = 0.301, current mu = 0.338) allocation failure 
[22685:0x4126fd0]    34775 ms: Scavenge 2044.6 (2051.6) -> 2043.9 (2053.6) MB, 5.5 / 0.0 ms  (average mu = 0.301, current mu = 0.338) allocation failure 


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x140de99]
    1: StubFrame [pc: 0x138f740]
Security context: 0x3f3e630c08d1 <JSObject>
    2: addJSDocComment(aka addJSDocComment) [0x366908a85049] [/home/lmcmcds/theia/node_modules/typescript/lib/typescript.js:~19860] [pc=0x1459a05c4b35](this=0x39c986dc04b1 <undefined>,0x255d8d2debf1 <NodeObject map = 0x346782279469>)
    3: parseTypeMember(aka parseTypeMember) [0x366908a87449] [/home/lmcmcds/theia/node_modules/typ...

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0xa1a640 node::Abort() [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
 2: 0xa1aa4c node::OnFatalError(char const*, char const*) [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
 3: 0xb9a68e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
 4: 0xb9aa09 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
 5: 0xd57c85  [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
 6: 0xd58316 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
 7: 0xd64bd5 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
 8: 0xd65a85 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
 9: 0xd6853c v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
10: 0xd2ef5b v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
11: 0x107158e v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
12: 0x140de99  [/home/lmcmcds/.nvm/versions/node/v12.22.1/bin/node]
Aborted (core dumped)
```

@vince-fugnitto can you give this a quick try?

**Comment by marcdumais-work** [comment]
Ah, just noticed you cover that in the description. How is it handled in CI?

**Comment by marcdumais-work** [comment]
> Ah, just noticed you cover that in the description. How is it handled in CI?

Ah, part of CI's env!

**Review by marcdumais-work — APPROVED** [review]
LGTM, thanks @paul-marechal

**Comment by tsmaeder** [comment]
@paul-marechal it seems we're not getting any doc in @theia/plugin anymore: maybe this line is to blame? https://github.com/eclipse-theia/theia/pull/10182/files#diff-4b9912b01bbc075872cc9df6fa2f44c711c4f438a13ce271a7a6efcf3a1088fcL6

Do we need to raise an issue?

**Comment by marcdumais-work** [comment]
> Do we need to raise an issue?

Yes, please do.

**Comment by tsmaeder** [comment]
https://github.com/eclipse-theia/theia/issues/10227

**Pull request merged**